src/main/kotlin/org/gradle/script/lang/kotlin/provider/KotlinBuildScriptCompiler.kt

```diff
@@ -59,9 +59,9 @@ class KotlinBuildScriptCompiler(
     val gradleApiExtensions: ClassPath,
     gradleScriptKotlinJars: ClassPath) {
 
-    val scriptResource = scriptSource.resource!!
-    val scriptFile = scriptResource.file!!
-    val script = convertLineSeparators(scriptResource.text!!)
+    val scriptFilePath = scriptSource.fileName!!
+    val scriptFile = File(scriptFilePath)
```

Comment:
The `File` instance here is just a vehicule for "path" and "filename". It isn't used as a *file*.
When the file doesn't exists, it carries its intended path.

Given that we require Java 8 we could use `Path` instead to better convey the intention. But